internal/suite/setup.go

```diff
@@ -70,13 +74,41 @@ func Setup(isUpgrade bool, clusterBuilder cb.ClusterBuilder, clusterReadyFns ...
 				events, err := framework.MC().GetEventsForResource(ctx, clusterApp)
 				if err != nil {
 					logger.Log("Failed to get events for App: %v", err)
-					return
-				}
-				if len(events.Items) == 0 {
-					logger.Log("No events found for Cluster App")
+				} else {
+					if len(events.Items) == 0 {
+						logger.Log("No events found for Cluster App")
+					}
+					for _, event := range events.Items {
+						logger.Log("Event: Reason='%s', Message='%s', Last Occurred='%v'", event.Reason, event.Message, event.LastTimestamp)
+					}
```

Comment:
Just a question, because I'm not 100% into the code: Do we need the `else` here or could we also just omit it for readability? At least I don't get the difference as long as events is never `nil`. And if it would, checking only `err` doesn't feel sufficient to me.

Reply:
More for consistency with the other checks in this function. 

But because we want it to log out in both cases but not both today. If there is an error we want to let the user know about it but if there wasn't an error we want to display logs based on what we fetched.

Reply:
Ah, right! Makes sense! Otherwise we would end up with 2 lines of log for the same reason.